src/graph.rs

```diff
@@ -130,38 +139,29 @@ impl Range {
 
 #[derive(Debug, Clone)]
 pub enum ModuleGraphError {
-  InvalidTypeAssertion {
-    specifier: ModuleSpecifier,
-    range: Range,
-    actual_media_type: MediaType,
-    expected_media_type: MediaType,
-  },
   LoadingErr(ModuleSpecifier, Option<Range>, Arc<anyhow::Error>),
   Missing(ModuleSpecifier, Option<Range>),
   ParseErr(ModuleSpecifier, deno_ast::Diagnostic),
-  ResolutionError(ResolutionError),
-  UnsupportedImportAssertionType {
-    specifier: ModuleSpecifier,
-    range: Range,
-    kind: String,
-  },
   UnsupportedMediaType {
     specifier: ModuleSpecifier,
     media_type: MediaType,
     maybe_referrer: Option<Range>,
   },
```

Comment:
RE https://github.com/denoland/deno_graph/pull/228#discussion_r1103865391 -- I wonder if we should have `ModuleError(ModuleError)` then put `LoadingErr`, `Missing`, `ParseError`, and `UnsupportedMediaType` in there. Then we could move `pub fn specifier(&self) -> &ModuleSpecifier` onto that and then move `maybe_range` into the other errors. What do you think?

Reply:
Yeah that's what I did in the previous attempt from last year I mentioned https://github.com/nayeemrmn/deno_graph/blob/786128a3f6330b1bdf7c6408e2ca53eb9c4688e9/src/graph.rs#L297-L302. I'll try it again, it should go better with this version's solution.